File: backend-jakarta/src/main/java/com/example/backendjakarta/dao/MenuItemDao.java
package com.example.backendjakarta.dao;

import com.example.backendjakarta.SQLService;
import com.example.backendjakarta.models.MenuItemModel;
import lombok.SneakyThrows;

import java.sql.*;
import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

public class MenuItemDao {

    @SneakyThrows
    public static void create(MenuItemModel item) {
        String sql = "INSERT INTO menu_item (name, price) VALUES (?, ?)";
        try (PreparedStatement stmt = SQLService.getConnection().prepareStatement(sql)) {
            stmt.setString(1, item.getName());
            stmt.setBigDecimal(2, item.getPrice());
            stmt.executeUpdate();
        }
    }

    @SneakyThrows
    public static MenuItemModel read(int id) {
        String sql = "SELECT * FROM menu_item WHERE id = ?";
        try (PreparedStatement stmt = SQLService.getConnection().prepareStatement(sql)) {
            stmt.setInt(1, id);
            try (ResultSet rs = stmt.executeQuery()) {
                if (rs.next()) {
                    String name = rs.getString("name");
                    BigDecimal price = rs.getBigDecimal("price");
                    return new MenuItemModel(id, name, price);
                }
            }
        }
        return null;
    }

    @SneakyThrows
    public static void update(MenuItemModel item) {
        String sql = "UPDATE menu_item SET name = ?, price = ? WHERE id = ?";
        try (PreparedStatement stmt = SQLService.getConnection().prepareStatement(sql)) {
            stmt.setString(1, item.getName());
            stmt.setBigDecimal(2, item.getPrice());
            stmt.setInt(3, item.getId());
            stmt.executeUpdate();
        }
    }

    @SneakyThrows
    public static void delete(int id) {
        String sql = "DELETE FROM menu_item WHERE id = ?";
        try (PreparedStatement stmt = SQLService.getConnection().prepareStatement(sql)) {
            stmt.setInt(1, id);
            stmt.executeUpdate();
        }
    }


    @SneakyThrows
    public static List<MenuItemModel> findAll() {
        List<MenuItemModel> items = new ArrayList<>();
        String sql = "SELECT * FROM menu_item";
        try (Statement stmt = SQLService.getConnection().createStatement();
             ResultSet rs = stmt.executeQuery(sql)) {
            while (rs.next()) {
                int id = rs.getInt("id");
                String name = rs.getString("name");
                BigDecimal price = rs.getBigDecimal("price");
                MenuItemModel item = new MenuItemModel(id, name, price);
                items.add(item);
            }
        }
        return items;
    }
}

File: backend-servlet/src/main/kotlin/com/example/dao/MenuItemDao.kt
package com.example.dao

import com.example.SqlService
import com.example.models.MenuItemModel
import java.math.BigDecimal


object MenuItemDao {
    fun create(item: MenuItemModel) {
        val sql = "INSERT INTO menu_item (name, price) VALUES (?, ?)"
        SqlService.getConnection().prepareStatement(sql).use { stmt ->
            stmt.setString(1, item.name)
            stmt.setBigDecimal(2, item.price)
            stmt.executeUpdate()
        }
    }
    
    fun read(id: Int): MenuItemModel? {
        val sql = "SELECT * FROM menu_item WHERE id = ?"
        SqlService.getConnection().prepareStatement(sql).use { stmt ->
            stmt.setInt(1, id)
            stmt.executeQuery().use { rs ->
                if (rs.next()) {
                    val name: String = rs.getString("name")
                    val price: BigDecimal = rs.getBigDecimal("price")
                    return MenuItemModel(id, name, price)
                }
            }
        }
        return null
    }
    
    fun update(item: MenuItemModel) {
        val sql = "UPDATE menu_item SET name = ?, price = ? WHERE id = ?"
        SqlService.getConnection().prepareStatement(sql).use { stmt ->
            stmt.setString(1, item.name)
            stmt.setBigDecimal(2, item.price)
            stmt.setInt(3, item.id)
            stmt.executeUpdate()
        }
    }
    
    fun delete(id: Int) {
        val sql = "DELETE FROM menu_item WHERE id = ?"
        SqlService.getConnection().prepareStatement(sql).use { stmt ->
            stmt.setInt(1, id)
            stmt.executeUpdate()
        }
    }
    
    fun findAll(): List<MenuItemModel> {
        val items: MutableList<MenuItemModel> = ArrayList<MenuItemModel>()
        val sql = "SELECT * FROM menu_item"
        SqlService.getConnection().createStatement().use { stmt ->
            stmt.executeQuery(sql).use { rs ->
                while (rs.next()) {
                    val id: Int = rs.getInt("id")
                    val name: String = rs.getString("name")
                    val price: BigDecimal = rs.getBigDecimal("price")
                    val item = MenuItemModel(id, name, price)
                    items.add(item)
                }
            }
        }
        return items
    }
}

